spec/rubocop/cop/rails/root_pathname_methods_spec.rb

```diff
@@ -191,4 +191,10 @@
       file = Rails.root.join('docs', 'invoice.pdf').open
     RUBY
   end
+
+  it 'does not register an offense when using `File.join(Rails.root, ...)`' do
+    expect_no_offenses(<<~RUBY)
+      File.join(Rails.root, '/foo')
```

Comment:
The detection should occur unless the 2nd argument+ of `File.join` starts with a `/`.

```ruby
Pathname.new('/foo').join('/bar').to_s # unexpected "/bar"
Pathname.new('/foo').join('bar').to_s  # expected "/foo/bar"
```

See for details: https://bugs.ruby-lang.org/issues/19155